eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java

```diff
@@ -326,25 +334,27 @@ private boolean checkIfExistingRSOrCandidateForMerge(SubmittedVariantEntity subm
     }
 
     private void checkIfCandidateForRSSplit(SubmittedVariantEntity submittedVariantEntity,
-                                               ClusteredVariantEntity clusteredVariantEntity,
-                                               String assembly,
-                                               Map<Long, Set<String>> allExistingHashesGroupByRS,
-                                               Map<Long, SubmittedVariantOperationEntity> rsSplitCandidateSVOE) {
+                                            ClusteredVariantEntity clusteredVariantEntity,
+                                            String assembly,
+                                            Map<Long, Set<String>> allExistingHashesGroupByRS,
+                                            Map<Long, SubmittedVariantOperationEntity> rsSplitCandidateSVOE,
+                                            Map<Long, SubmittedVariantOperationEntity>  updateRsSplitCandidateSVOE) {
         Long variantAccession = clusteredVariantEntity.getAccession();
         String variantHash = clusteredVariantEntity.getHashedMessage();
 
         if (!allExistingHashesGroupByRS.get(variantAccession).isEmpty() &&
                 !allExistingHashesGroupByRS.get(variantAccession).contains(variantHash)) {
             if (rsSplitCandidateSVOE.containsKey(variantAccession)) {
-                List<SubmittedVariantInactiveEntity> inactiveEntities = rsSplitCandidateSVOE.get(variantAccession)
-                        .getInactiveObjects();
+                SubmittedVariantOperationEntity submittedVariantOperationEntity = rsSplitCandidateSVOE.get(variantAccession);
+                List<SubmittedVariantInactiveEntity> inactiveEntities = submittedVariantOperationEntity.getInactiveObjects();
                 boolean submittedVariantAlreadyExist = inactiveEntities.stream()
                         .anyMatch(sv -> submittedHashingFunction.apply(sv).equals(
                                 submittedVariantEntity.getHashedMessage()) &&
                                 sv.getAccession().equals(submittedVariantEntity.getAccession()));
                 if (!submittedVariantAlreadyExist) {
                     inactiveEntities.add(new SubmittedVariantInactiveEntity(submittedVariantEntity));
                 }
+                updateRsSplitCandidateSVOE.put(variantAccession, submittedVariantOperationEntity);
```

Comment:
This is present in the if and else statements, can be taken out so it is not repeated

Reply:
Done